themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,371 @@
+
+What you really need, then, is to be able to take multiple independent, possibly long-running actions based on a single API Gateway request. And one way to do that is with [Amazon EventBridge](https://aws.amazon.com/eventbridge).
+
+## Hello, EventBridge
+
+Amazon EventBridge (formerly CloudWatch Events) is as a serverless [_event bus_](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-bus.html) whose job is to receive structured [_event data_](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html) --- from your own applications, from other AWS Services --- and use that data to notify other applications or services using event-handling [_rules_](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-rules.html) that you specify. With EventBridge, you can build loosely coupled, event-driven systems that take advantage of AWS's rich support for serverless architectures and that scale gracefully as the needs of those systems change over time.
+
+For this particular application, EventBridge would let you address the multiple-handler problem in a more scalable and easily maintainable way. Rather than have API Gateway invoke Lambda directly, leaving one Lambda responsible for handling multiple tasks, you could have API Gateway publish to EventBridge instead, and let EventBridge invoke as many Lambdas (or other AWS services) as you like, all serverlessly and in parallel --- and of course, all easily managed with Pulumi.
+
+Let's see how. Here's a revised architecture diagram showing how you might approach building an application like this one with EventBridge positioned between API Gateway and Lambda:
+
+![Figure 3: Using EventBridge](./figure-3.png)
+
+Now let's have a look at what it'd be like to build it with Pulumi. We won't build _everything_ in this diagram --- things like writing to the database or messaging Slack are left for you to explore --- but we will build enough to give you clear picture and a working example of how to connect all of these parts into a working application. Specifically:
+
+* an API Gateway [instance](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-vs-rest.html) to act as a container for your public API, along with a [_stage_](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-stages.html) and a [_route_](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-routes.html) to handle inbound HTTP requests;
+* an EventBridge [integration](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations.html) (comprised of an EventBus and EventRule) to handle notifications from API Gateway, and finally;
```

Comment:
Makes sense, yeah. I'll lowercase here, though, since in these bullets, I'm describing AWS resources more than the actual Pulumi ones (which come later, in the code).